packages/core/src/messages/responses/list-indexes.ts

```diff
@@ -0,0 +1,66 @@
+import {SdkError} from '../../errors';
+import {ResponseBase, ResponseError, ResponseSuccess} from './response-base';
+
+/**
+ * Parent response type for a list indexes request.  The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Success}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof ListIndexes.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `ListIndexes.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+export abstract class Response extends ResponseBase {}
+
+class _Success extends Response {
+  private readonly indexes: string[];
+  constructor(indexes: string[]) {
+    super();
+    this.indexes = indexes;
+  }
+
+  /**
+   * An array of index names.
+   * @returns {string[]}
+   */
+  public getIndexes() {
+    return this.indexes;
+  }
```

Comment:
To align with the protos

```suggestion
  public getIndexNames() {
    return this.indexNames;
  }
```